x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/AbstractBlock.java

```diff
@@ -45,7 +45,6 @@ protected AbstractBlock(int positionCount, @Nullable int[] firstValueIndexes, @N
         this.blockFactory = blockFactory;
         this.firstValueIndexes = firstValueIndexes;
         this.nullsMask = nullsMask == null || nullsMask.isEmpty() ? null : nullsMask;
-        assert (firstValueIndexes == null && this.nullsMask == null) == false;
```

Comment:
This assertion prevents me from creating a valid Block when both nullsMask and firstValueIndexes are null. I've removed this assertion. I think the correct one is `nullMask != null -> firstValueIndexes == null || firstValueIndexes.length == positionCount`. I will need to modify some tests before enabling that assertion, and I will do this in a follow-up.

Reply:
If there aren't any nulls or multivalued fields shouldn't we make a `Vector` instead?

Reply:
Thanks, I restored the assertion with the message in https://github.com/elastic/elasticsearch/pull/100465/commits/97760a7ff6a791552e2369af16853f700ba817a1.